src/compiler/moduleNameResolver.ts

```diff
@@ -1047,9 +1049,18 @@ namespace ts {
     }
 
     function loadModuleFromPackageJson(jsonContent: PackageJsonPathFields, extensions: Extensions, candidate: string, failedLookupLocations: Push<string>, state: ModuleResolutionState): PathAndExtension | undefined {
-        const file = tryReadPackageJsonFields(extensions !== Extensions.JavaScript && extensions !== Extensions.Json, jsonContent, candidate, state);
+        let file = tryReadPackageJsonFields(extensions !== Extensions.JavaScript && extensions !== Extensions.Json, jsonContent, candidate, state);

```

Comment:
Nit: might just modify `tryReadPackageJsonFields` to take separate `readTypes` and `readMain` parameters, if we might want both at once.

Reply:
In most both cases different conditions determine if we read main types so i would rather leave it as is.